### PR TITLE
fixes #829, SelectExpandQueryValidator throws when $selecting complex property

### LINKED ISSUE #829: SelectExpandQueryValidator throws when $selecting complex property (EDM loaded from CSDL)
Using the latest on main (trying to debug the other issues I've run into in released versions)
> Unable to cast object of type 'Microsoft.OData.Edm.Csdl.CsdlSemantics.CsdlSemanticsNamedTypeReference' to type 'Microsoft.OData.Edm.IEdmComplexTypeReference'."
at Microsoft.AspNetCore.OData.Query.Validator.SelectExpandQueryValidator.GetStructuredType(IEdmTypeReference typeRef)
   at Microsoft.AspNetCore.OData.Query.Validator.SelectExpandQueryValidator.ValidatePathSelectItem(PathSelectItem pathSelectItem, SelectExpandValidatorContext validatorContext)
   at Microsoft.AspNetCore.OData.Query.Validator.SelectExpandQueryValidator.ValidateSelectExpand(SelectExpandClause selectExpandClause, SelectExpandValidatorContext validatorContext)
   at Microsoft.AspNetCore.OData.Query.Validator.SelectExpandQueryValidator.Validate(SelectExpandQueryOption selectExpandQueryOption, ODataValidationSettings validationSettings)
   at Microsoft.AspNetCore.OData.Query.SelectExpandQueryOption.Validate(ODataValidationSettings validationSettings)
   at Microsoft.AspNetCore.OData.Query.Validator.ODataQueryValidator.Validate(ODataQueryOptions options, ODataValidationSettings validationSettings)
   at Microsoft.AspNetCore.OData.Query.ODataQueryOptions.Validate(ODataValidationSettings validationSettings)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.ValidateQuery(HttpRequest request, ODataQueryOptions queryOptions)
   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.OnActionExecuting(ActionExecutingContext actionExecutingContext)

This error does not occur in the released 8.0.12 version. Note that `GetStructuredType` seems to be superfluous as odata.net has an extension method `.ToStructuredType()` which works correctly in this case.

### PR DESCRIPTION
**Pull request by xuzhg**
fixes #829.

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml [hunk 1]
```diff
@@ -14827,3 +14813,19 @@
         </member>
     </members>
 </doc>
+ummary>
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml:14816]: That problem of truncated `<summary>` again...

File: src/Microsoft.AspNetCore.OData/Query/Validator/SelectExpandQueryValidator.cs [hunk 2]
```diff
@@ -280,7 +280,7 @@ protected virtual void ValidatePathSelectItem(PathSelectItem pathSelectItem, Sel
                 }
 
                 property = propertySegment.Property;
-                structuredType = GetStructuredType(propertySegment.Property.Type);
+                structuredType = propertySegment.Property.Type.ToStructuredType();
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Query/Validator/SelectExpandQueryValidator.cs:283]: Is it possible to add a test to demonstrate that an exception is not thrown after this change?

Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/Query/Validator/SelectExpandQueryValidator.cs:283]: ToStructuredType is an extension method in ODL , this method is well-tested at ODL , the implementation doesn't not throw exception.

**Review by julealgon — DISMISSED** [review]

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml]: #Resolved.

**Comment by xuzhg** [comment]
@gathogojr I'd like to merge it and move forward it.

**Pull request merged**